projects/swimlane/ngx-ui/src/lib/components/json-editor/node-types/array-node.component.html

```diff
@@ -4,10 +4,7 @@
       <ngx-dropdown [showCaret]="true">
         <ngx-dropdown-toggle>
           <div class="type-icon" tooltipPlacement="top">
-            <ngx-icon
-              [fontIcon]="dataTypeMap[(schemas[i]?.type)] ? dataTypeMap[(schemas[i]?.type)].icon : 'integration'"
-            >
-            </ngx-icon>
+            <ngx-icon [fontIcon]="getIcon(schemas[i])"> </ngx-icon>
```

Comment:
Can we do something so we don´t call this function all the time? Perhaps have the "fontIcon" stored in a public variable in the component, and only call getIcon() on OnChanges?

Reply:
Yup, I am caching them in the schema $meta.icon property now.